pkg/operation/botanist/garbage_collection.go

```diff
@@ -95,3 +102,41 @@ func (b *Botanist) deleteStalePods(client kubernetes.Interface, podList *corev1.
 
 	return result
 }
+
+func (b *Botanist) removeStaleOutOfDiskNodeCondition() error {
+	// TODO: This code can be limited to 1.13.0-1.13.3 (assuming 1.13.3 has contains the Kubernetes fix) as soon as
+	// a new release containing the fix has been cut.
+	k8sVersionAtLeast113, err := utils.CompareVersions(b.Shoot.Info.Spec.Kubernetes.Version, ">=", "1.13")
+	if err != nil {
+		return err
+	}
+	if !k8sVersionAtLeast113 {
+		return nil
+	}
+
+	var nodeList *corev1.NodeList
+	if err := b.K8sShootClient.Client().List(context.TODO(), nil, nodeList); err != nil {
+		return err
+	}
+
+	var result error
+	for _, node := range nodeList.Items {
+		var (
+			conditions []corev1.NodeCondition
+			nodeCopy   = node.DeepCopy()
+		)
+
+		for _, condition := range node.Status.Conditions {
+			if condition.Type != corev1.NodeOutOfDisk {
+				conditions = append(conditions, condition)
+			}
+		}
+		nodeCopy.Status.Conditions = conditions
```

Comment:
Shouldn't we do a quick check if `len(conditions) != len(nodeCopy.Status.Conditions)` before we do the update? Then we can reduce no-ops.